File: application/src/main/java/whenyourcar/application/converter/user/UserCommonConverter.java
package whenyourcar.application.converter.user;

import whenyourcar.domain.entity.User;
import org.springframework.stereotype.Component;
import whenyourcar.application.dto.user.common.UserCommonResponse;

@Component
public class UserCommonConverter {
    public UserCommonResponse.UserProfileResponseDto toUserProfileResponse(User user){
        return UserCommonResponse.UserProfileResponseDto.builder()
                .userId(user.getId())
                .name(user.getName())
                .picture(user.getPicture())
                .role(user.getRole())
                .build();
    }

}

File: application/src/main/kotlin/howmuchcar/application/converter/user/UserCommonConverter.kt
package howmuchcar.application.converter.user

import howmuchcar.application.dto.user.UserProfileResponse
import howmuchcar.domain.entity.User
import org.springframework.stereotype.Component

@Component
class UserCommonConverter {
    fun toUserProfileResponse(user: User): UserProfileResponse {
        return UserProfileResponse(
            userId = user.id,
            name = user.name,
            picture = user.picture,
            email = user.email
        )
    }
}